modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy

```diff
@@ -647,42 +647,104 @@ class PodSpecBuilder {
         return res
     }
 
+
     @PackageScope
-    String getAcceleratorType(AcceleratorResource accelerator) {
+    void validateAccelerator(AcceleratorResource accelerator) {
+        // gpu-small: nvidia-t4 (1)
+        // gpu-large: nvidia-a10g (1)
+        // v100-x1: nvidia-v100 (1)
+        // v100-x4: nvidia-v100 (4)
+        // v100-x8: nvidia-v100 (8)
+
+        if (accelerator.type == null) {
+            accelerator.setProperty("type", "nvidia-t4")
+            log.info("No GPU type specified - defaulting to \"nvidia-t4\"")
+        }
 
-        def type = accelerator.type ?: 'nvidia.com'
+        if (
+            (accelerator.type == "nvidia-t4" && accelerator.limit == 1)
+            || (accelerator.type == "nvidia-a10g" && accelerator.limit == 1)
```

Comment:
maybe we want to allow limit to be 0 and then we don't provision? In case someone has workflow where they use a closure to define their GPU requirements where they set gpu limit to 0 if they don't want GPUs
not sure if that is valid nextflow tho, might want to double check